Clarify CDN details for PaaS vs SaaS (#368)
commerce-docs/microsite-commerce-storefront: src/content/docs/sdk/reference/initializer.mdx

## `setImageParamKeys(params)`

The `setImageParamKeys` method is part of the initializers module in the `@dropins/tools` package. It allows you to set image parameters globally for all drop-in components that use the [Image](/sdk/components/image/) component. CDN-based image optimization services, such as Fastly and Cloudflare, rely on these parameters to optimize images.

### Default behavior

Since Fastly is the default CDN for Adobe Commerce on Cloud, drop-in components add the following Fastly-style parameters to all image URLs by default:

- `width`
- `height`
- `auto`
- `quality`
- `crop`
- `fit`

Example:

```text
/media/catalog/product/adobestoredata/ADB150.jpg?auto=webp&quality=80&crop=false&fit=cover&width=960&height=1191`
```

Technical details about CDN-based image optimization differ depending on your backend Adobe Commerce solution:

> - <Badge tooltip="Applies to Adobe Commerce on Cloud projects (Adobe-managed PaaS infrastructure) and on-premises projects only." text="PaaS only" variant="note" href="https://experienceleague.adobe.com/en/docs/commerce/user-guides/product-solutions" /> Fastly is the default CDN-based image optimization service for Adobe Commerce on Cloud projects. Fastly-style parameters are used if `setImageParamKeys` is not called or if no parameters are provided.

Reviewer: ```suggestion
- <Badge tooltip="Applies to Adobe Commerce on Cloud projects (Adobe-managed PaaS infrastructure) and on-premises projects only." text="PaaS only" variant="note" href="https://experienceleague.adobe.com/en/docs/commerce/user-guides/product-solutions" /> Fastly is the default CDN-based image optimization service for Adobe Commerce on Cloud projects. Fastly-style parameters are used by default unless overridden via `setImageParamKeys`.
```

> - <Badge tooltip="Applies to Adobe Commerce as a Cloud Service and Adobe Commerce Optimizer projects only (Adobe-managed SaaS infrastructure)." text="SaaS only" variant="success" href="https://experienceleague.adobe.com/en/docs/commerce/user-guides/product-solutions" /> Cloudflare is the default CDN-based image optimization service for Adobe Commerce as a Cloud Service and Adobe Commerce Optimizer projects. Fastly-style parameters are used but are proxied through a [Cloudflare Worker](https://developers.cloudflare.com/images/transform-images/transform-via-workers) to transform the parameters to Cloudflare-style parameters if `setImageParamKeys` is not called or if no parameters are provided.

Reviewer: ```suggestion
- <Badge tooltip="Applies to Adobe Commerce as a Cloud Service and Adobe Commerce Optimizer projects only (Adobe-managed SaaS infrastructure)." text="SaaS only" variant="success" href="https://experienceleague.adobe.com/en/docs/commerce/user-guides/product-solutions" /> Cloudflare is the default CDN-based image optimization service for Adobe Commerce as a Cloud Service and Adobe Commerce Optimizer projects. Fastly-style parameters are used but are proxied through a [Cloudflare Worker](https://developers.cloudflare.com/images/transform-images/transform-via-workers) to transform the parameters to Cloudflare-style parameters. Overriding the default parameters using `setImageParamKeys` may interfere with the proxy transformation and lead to unexpected results.

### Default Behavior

- <Badge tooltip="Applies to Adobe Commerce on Cloud projects (Adobe-managed PaaS infrastructure) and on-premises projects only." text="PaaS only" variant="note" href="https://experienceleague.adobe.com/en/docs/commerce/user-guides/product-solutions" /> Fastly parameters are used if `setImageParamKeys` is not called or if no parameters are provided.

> - <Badge tooltip="Applies to Adobe Commerce as a Cloud Service and Adobe Commerce Optimizer projects only (Adobe-managed SaaS infrastructure)." text="SaaS only" variant="success" href="https://experienceleague.adobe.com/en/docs/commerce/user-guides/product-solutions" /> A Cloudlare worker rewrites the default Fastly parameters to a format accepted by the Cloudflare CDN if `setImageParamKeys` is not called or if no parameters are provided.

Author: @fnhipster, I'm not sure if we even want to talk about Cloudflare specifically. I only proposed this change so that I could comment on it in this PR and get feedback from the team.

Reviewer 2: I don't even think this is true. I thought the storefront code just renders the image urls from the API/backend, but with some Fastly parameters (because we wrote it originally to support Fastly only).

Is this "rewrites the default Fastly parameters..." new behavior? I am fresh back from PTO so apologies if this is obvious, but it doesn't make sense to me.

Author: @sirugh, it's very possible that I misinterpreted from the source thread. It was loooong and a bit confusing 😅 

Hoping to get clarity/confirmation in this review.

Reviewer: To clarify, Drop-ins add the following parameters to all image URLs by default:

- `width`  
- `height`  
- `auto`  
- `quality`  
- `crop`  
- `fit`

**Example:**  
`/media/catalog/product/adobestoredata/ADB150.jpg?auto=webp&quality=80&crop=false&fit=cover&width=960&height=1191`

CDN-based image optimization services, such as Fastly, use these parameters. Adobe Commerce itself does not handle image optimization; the CDN performs this. We use Fastly-style parameters because Fastly is the CDN bundled with Adobe Commerce.

That said, merchants can use other CDNs with different optimization services that require different parameters or URL patterns. In such cases, they can override the defaults using the `setParamsKeys` function to match their specific needs.

For more complex URL formats—such as Cloudflare’s: `https://example.com/cdn-cgi/image/format=webp,quality=80,width=960,height=1191/your/image.jpg`— these can be proxied through a Cloudflare Worker, as implemented by Vijay Krishna Dandu.

Author: Thanks for the review @fnhipster & @sirugh. I pushed some changes to address. How do they look? Any concerns?
